src/main/java/org/pih/petl/job/SqlServerImportJob.java

```diff
@@ -136,6 +137,64 @@ else if (StringUtils.isEmpty(partitionColumn)) {
             throw new PetlException("You must specify a value for your partition column");
         }
 
```

Comment:
In a Bulk Load job, one can make things "incremental" by taking the following steps:

1. The "previousWatermarkQuery" retrieves the low water mark for any updates.  This query should return the datetime for which any records last changed on or before this datetime are determined to be up-to-date and don't need to be reloaded.  This is evaluated to a previousWatermark datetime.

2. The "newWatermarkQuery" retrieves the high water mark for any updates.  Any records who have last changed _after_ this datetime are considered too new and not processed as part of this incremental loading job.  This is to ensure consistency between records being queried on the source system and target system.  This is evaluated to a newWatermark datetime.

3. The "incrementalDeleteQuery" represents the statement that should be run to delete any records that are considered changed during the execution window (between the previousWatermark and newWatermark).

4. The "schema" sql statement for loading needs to be modified to contain a columns that identify which rows need to be deleted and re-inserted.  This may typically be a column for patient_id (if changes are determined by patient-level change events) and a column for last_updated datetime.

5. The "extract" sql query for querying MySQL needs to be modified to return all rows if previousWatermark is null, or to return only a subset of rows if previousWatermark is not null.  This needs to be consistent with the definition of the incrementalDeleteQuery, to ensure that any rows that are deleted are re-inserted if they are still valid.

The principle is this, if incremental.enabled = true

1. A new partition is created as normal
2. The watermark values are retrieved
3. The partition is pre-loaded with existing data
4. The incrementalDeleteQuery is used to remove any rows that have been changed in the watermark window
5. The bulk load operation happens as normal, but only inserting the subset of rows that have changed in the watermark window

This means that over time, any data that was previously loaded that is subsequently deleted or voided in the source system will be deleted out of the target warehouse table, since it will be deleted and not re-inserted.  And it means that we don't need any complexity of updates vs. inserts of existing data since updates are just treated as deletes + inserts.

This will be prone to some user error as the configuration and SQL statements need to be carefully crafted and tweaked, but there is copious logging added to help ensure that implementers can debug as they craft their configuration.